app/jobs/application_job.rb

```diff
@@ -53,3 +105,35 @@ def zip_info_check!(metadata)
     end
   end
 end
+
+# Q: Why copy the logic out of resque-lock instead of using the gem?
+# A: It's a small amount of (MIT licensed) code, the gem is and unmaintained (last updated in 2012),
+# and the lock cleanup hook wasn't getting invoked automatically in our jobs.  Inlining the code
+# reduces indirection and dependency (on both resque-lock and on Resque itself, since we use ActiveJob
+# hooks here instead of Resque hooks).
```

Comment:
I think in the future, no one will remember there was a resque-lock gem.  I think adding "gem" after resque-lock below might be sufficient, and leave lines 109-113 out.   If you're attached to the above, that's fine too.